include/internal/refcount.h

```diff
@@ -48,7 +54,11 @@ static inline int CRYPTO_UP_REF(CRYPTO_REF_COUNT *refcnt, int *ret)
  */
 static inline int CRYPTO_DOWN_REF(CRYPTO_REF_COUNT *refcnt, int *ret)
 {
+#   ifdef OSSL_TSAN_BUILD
+    *ret = atomic_fetch_sub_explicit(&refcnt->val, 1, memory_order_acq_rel) - 1;
+#   else
```

Comment:
There is no reason for acquire+release, we want just release. And we want it for all versions, not just a tsan build.

Reply:
I think @dot-asm arguments in #6900 are sound.

The acq+rel is apparently necessary for TSAN to be silent according to #25660

Reply:
Can you at least add some comment about tsan requiring that?